### DISCUSSION TITLE
I can't install the plugin version 1.5.0

### DISCUSSION THREAD

**Discussion by pmarchal25** [Onemind-Services-LLC/netbox-secrets#76] [General]
Hi, i have netbox 3.3.10 and i would like to install netbox-secrets 1.5.0 or 1.4.1 in my netbox version. The problem is that when i try to install the plugin, it appears an error called  **ModuleNotFoundError: No module named 'importlib.util'** and i have tried to install this module but i can't. What are the requirements for the instalation? Mb I need python 3.9 or something different?

To install the plugin i am in my python environment and i run sudo pip3 install netbox-secretstore==1.4.1 and the error **ModuleNotFoundError** appears.

Thanks,

Pedro

**Comment by abhi1693** [comment]
You need to install netbox-secretstore within your virtualenv. If you are already doing this, please share the full stackstrace of the issue

**Reply by pmarchal25** [reply]
Hi abhi1693, i have tried it but the same error appears. I have checked my python version and i am running the correct version (3.8.10). In addition, i have checked the directory /usr/lib/python3.8/ and the importlib appears there correctly. The full error is the following:

ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-zwdr6j0j/importlib/setup.py'"'"'; __file__='"'"'/tmp/pip-install-zwdr6j0j/importlib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-zwdr6j0j/importlib/pip-egg-info
         cwd: /tmp/pip-install-zwdr6j0j/importlib/
    Complete output (35 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 5, in <module>
        import distutils.core
      File "/usr/lib/python3.8/distutils/core.py", line 16, in <module>
        from distutils.dist import Distribution
      File "/usr/lib/python3.8/distutils/dist.py", line 19, in <module>
        from distutils.util import check_environ, strtobool, rfc822_escape
      File "/usr/lib/python3.8/distutils/util.py", line 9, in <module>
        import importlib.util
    ModuleNotFoundError: No module named 'importlib.util'
    Error in sys.excepthook:
    Traceback (most recent call last):
      File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 72, in apport_excepthook
        from apport.fileutils import likely_packaged, get_recent_crashes
      File "/usr/lib/python3/dist-packages/apport/__init__.py", line 5, in <module>
        from apport.report import Report
      File "/usr/lib/python3/dist-packages/apport/report.py", line 13, in <module>
        import fnmatch, glob, traceback, errno, sys, atexit, imp, stat
      File "/usr/lib/python3.8/imp.py", line 19, in <module>
        from importlib._bootstrap import _ERR_MSG, _exec, _load, _builtin_from_name
    ModuleNotFoundError: No module named 'importlib._bootstrap'

    Original exception was:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 5, in <module>
        import distutils.core
      File "/usr/lib/python3.8/distutils/core.py", line 16, in <module>
        from distutils.dist import Distribution
      File "/usr/lib/python3.8/distutils/dist.py", line 19, in <module>
        from distutils.util import check_environ, strtobool, rfc822_escape
      File "/usr/lib/python3.8/distutils/util.py", line 9, in <module>
        import importlib.util
    ModuleNotFoundError: No module named 'importlib.util'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.


Can you find a solution or maybe know what can be the problem?

Thank you so much.
Pedro

**Reply by abhi1693** [reply]
Do you not run NetBox within any virtual environment? Because if not, I'm not sure where the issue may be coming from then.

**Reply by pmarchal25** [reply]
I run netbox in my venv, i have installed netbox in opt/netbox directory, but i run it in the venv. I have this error when i try to install the plugin in the venv: Can not execute `setup.py` since setuptools is not available in the build environment.

**Reply by pmarchal25** [reply]
Hi again, now it appears this error: 
  Using cached importlib-1.0.4.zip (7.1 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      ERROR: Can not execute `setup.py` since setuptools is not available in the build environment.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

I have tested that setuptools is installed in the venv and the version is the last..However it doesn't work.. Do you know what can be the problem? I try to install this package from the venv using the command: pip3 install netbox-secretstore==1.4.0

Thank you so much.
Pedro
